lib/Driver/Driver.cpp

```diff
@@ -2031,6 +2031,12 @@ const ToolChain *Driver::getToolChain(const ArgList &Args) const {
       TC = new toolchains::Darwin(*this, Target);
       break;
     case llvm::Triple::Linux:
+      if (Target.isAndroid()) {
```

Comment:
I expect `os(Windows)` to be true for both Cygwin and a hypothetical port to the MSVC runtime and toolchains. Naming the toolchain `Cygwin` seems fine to me.